pkg/cvo/sync_worker.go

```diff
@@ -713,11 +716,18 @@ func equalSyncWork(a, b *SyncWork, context string) (equalVersion, equalOverrides
 
 // updateStatus records the current status of the sync action for observation
 // by others. It sends a copy of the update to the report channel for improved
-// testability.
-func (w *SyncWorker) updateStatus(update SyncWorkerStatus) {
+// testability. It does not overwrite payload load status and capabilities status
+// when payloadApplication is true since these statuses could be out-of-date.
+func (w *SyncWorker) updateStatus(update SyncWorkerStatus, payloadApplication bool) {
```

Comment:
Hmm, if we have payload-loaders and payload-appliers both feeding this method, couldn't one always have stale information about the other?  Can we set this up so they only push the information that they're responsible for, with pointers or other "I don't have an opinion" markers so they can clearly leave the properties they aren't responsible for alone?  Maybe `updateStatus(*loadStatus, *applyStatus)` or something like that?

Reply:
> Can we set this up so they only push the information that they're responsible

That's essentially what I have using the `bool`. But I had also thought about simply making another update method so I'll do that. I think it will be easier to understand than a `bool` or using `nil`s.